legend-engine-core/legend-engine-core-language-pure/legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/runtime/StoreProviderPointerFactory.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.language.pure.grammar.from.runtime;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.eclipse.collections.impl.utility.ListIterate;
+import org.finos.legend.engine.language.pure.grammar.from.PureGrammarParserUtility;
+import org.finos.legend.engine.protocol.pure.v1.model.SourceInformation;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PackageableElementType;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.StoreProviderPointer;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+public interface StoreProviderPointerFactory
+{
+    static StoreProviderPointer parseStoreProviderType(TerminalNode type)
+    {
+        StoreProviderPointer storeProviderPointer = new StoreProviderPointer();
+        if (Objects.isNull(type))
+        {
+            storeProviderPointer.type = PackageableElementType.STORE;
+        }
+        else
+        {
+            storeProviderPointer.type = PackageableElementType.valueOf(type.getText().toUpperCase());
+        }
+        return storeProviderPointer;
+    }
+
+    static StoreProviderPointer create(ParserRuleContext storeProviderPointerContext, SourceInformation sourceInformation)
```

Comment:
I rather keep the antlr parsing on the extension parsers and avoid some of the class name lookups...